### PR TITLE
bugfix: improve selection of webview

### PR DESCRIPTION
**Pull request by pollend**
I played with a couple of solutions; sometimes there is an id that links the view back to the containing webview but that is not consistent across vscode. The main issue is the webviews are elevated to the top of the dom so you can't easily tell if that webview is associated with a given element. Testing the overlap seems the most consistent way to approach this?

ref: https://github.com/redhat-developer/vscode-extension-tester/issues/1492

Before submitting your PR, please review the following checklist:

- [ ] **CONSIDER** adding a new test if your PR resolves an issue.
- [ ] **DO** keep pull requests small so they can be easily reviewed.
- [ ] **DO** make sure tests pass.
- [ ] **DO** make sure any public APIs changes are documented.
- [ ] **DO** make sure not to introduce any compiler warnings.

Before merging the PR:

- [ ] **CHECK** continuous integration of `main` branch is green.
- [ ] **CHECK** pull request check job is green.
- [ ] **CHECK** all pull request questions/requests are resolved.
- [ ] **WAIT** till PR is approved by at least 1 committer.

### CODE REVIEW

File: packages/page-objects/src/locators/locators.ts [hunk 1]
```diff
@@ -627,7 +627,7 @@ export async function findBestContainingElement(container: IRectangle, testEleme
 		const ax = Math.max(container.x, rect.x);
 		const ay = Math.max(container.y, rect.y);
 		const bx = Math.min(container.x + container.width, rect.x + rect.width);
-		const by = Math.min(container.y + container.width, rect.y + rect.height);
+		const by = Math.min(container.y + container.height, rect.y + rect.height);
```
Comment (pollend) [inline packages/page-objects/src/locators/locators.ts:630]: I made a small mistake here.

File: tests/test-project/src/test/webview/webviewView.test.ts [hunk 2]
```diff
@@ -52,7 +54,7 @@ describe('WebviewViews', function () {
 				this.timeout(10000);
 				await new Workbench().executeCommand(param.command);
 				await new Promise((res) => setTimeout(res, 2000));
-				webviewView = new WebviewView();
+				webviewView = new WebviewView(param.container());
```
Comment (pollend) [inline tests/test-project/src/test/webview/webviewView.test.ts:57]: the selection criteria will select Welcome area first if this test is run separately. not sure if this is better to make the selection criteria more exact?

**Comment by pollend** [comment]
Hello, @djelinek 

Is there anything else I would need to address? Any comment would be appreciated.

**Comment by djelinek** [comment]
Hello, 
thank you for your PR and time you have invested here, really appreciated! I will take a look closer ASAP this week.

From top of my head, it would be nice to also have test cases which are handling multiple WebView types opened at same time.

Basically it can happen state when, Editor WebView / BottomBar WebView and SideBar WebView are open/visible at same time. This is a edge case which was reported in a past also

**Comment by pollend** [comment]
> Hello, thank you for your PR and time you have invested here, really appreciated! I will take a look closer ASAP this week.
> 
> From top of my head, it would be nice to also have test cases which are handling multiple WebView types opened at same time.
> 
> Basically it can happen state when, Editor WebView / BottomBar WebView and SideBar WebView are open/visible at same time. This is a edge case which was reported in a past also

I have one test where i open multiple grouped webview but are you saying I should like open the sidebar and bottom section as well?

**Comment by djelinek** [comment]
> > Hello, thank you for your PR and time you have invested here, really appreciated! I will take a look closer ASAP this week.
> > From top of my head, it would be nice to also have test cases which are handling multiple WebView types opened at same time.
> > Basically it can happen state when, Editor WebView / BottomBar WebView and SideBar WebView are open/visible at same time. This is a edge case which was reported in a past also
> 
> I have one test where I open multiple grouped webview but are you saying I should add another test for BottomBar/Editor Webview and Editor Webview/sidebar

That is awesome what you have added, it brings a value. I am more expressing my first thoughts here about complexity of problem.

What I am saying is that we need to cover also case, mentioned above, with multiple webviews of different kind opened in a same time, probably in a new standalone test. It can be also checked manually if this is working and we can provide other test in another PR. I am just raising it to not forget about that 🙂

**Comment by pollend** [comment]
sorry, for the lack of response I've been on a bit of a trip. I'll see if I can address your comments by the end of this week.

**Comment by rajwcg** [comment]
I am eagerly waiting this issue to be fixed. I am facing an issue with multiple web views on the page—one in the sidebar and another in the editor. After the sidebar view is launched, I am unable to locate the editor web view. Could you please help me resolve this? I have already tried using the `openEditor` method from the web view, but it didn’t work.

**Comment by pollend** [comment]
Hello, @djelinek where have we left off, is there anything else that needs to be address?

**Comment by djelinek** [comment]
> Hello, @djelinek where have we left off, is there anything else that needs to be address?

Hello @pollend, big sorry for letting this PR hanging there. I was a bit struggling last couple of weeks.  I will address everything till EOW.

Once again big sorry and tahnky you for your patience.
